### PR TITLE
fix(#3): Fix relay shut down when only one metric is sent

### PR DESCRIPTION
**Pull request by senzzzi**
Should fix: https://github.com/statful/statful-relay/issues/3

### CODE REVIEW

File: lib/listeners/statful.js [hunk 1]
```diff
@@ -29,7 +19,7 @@ function dispatchMetricLine(listener) {
             });
 
             if (listener.config.stats) {
-                listener.statfulClient.counter('metrics_flushed', countMetricsInLines(lines));
+                listener.statfulClient.counter('metrics_flushed', linesList.length);
```
Comment (miguelcnf) [inline lib/listeners/statful.js]: We want to know the number of metrics flushed here - this way we're only measuring the length of the payload string.

File: lib/listeners/statful.js [hunk 2]
```diff
@@ -3,16 +3,6 @@ var Logger = require('../logger');
 var Utils = require('../utils');
 
 /**
- * Counts the number of lines in the passed metrics lines
- *
- * @param metricLines A string containing lines of metrics
- * @returns {number}
- */
-function countMetricsInLines(metricLines) {
-    return metricLines.match(/\n/gi).length + 1;
```
Comment (miguelcnf) [inline lib/listeners/statful.js]: We can check the return value of the [match](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/match) method here and if it is null then return 1.

Comment (miguelcnf) [inline lib/listeners/statful.js]: And also check that `metricLines` isn't an empty string of course.

File: lib/listeners/statful.js (hunk 1 above)
Comment (senzzzi) [inline lib/listeners/statful.js]: But the number of metrics flushed equals the number of lines in the payload string.. What's the difference between counting lines and counting "\n" and add 1?

Comment (miguelcnf) [inline lib/listeners/statful.js]: I'm sorry, you're right. I was misreading `linesList` with `lines` and assuming you were getting the length of the original string.

**Comment by miguelcnf** [comment]
@senzzzi can you please increment the package patch version before we merge this?

**Comment by miguelcnf** [comment]
Thanks for this @senzzzi.

`1.1.3` published to npm.

**Pull request merged**